### DISCUSSION TITLE
Merge is not happening in organization account via Github API

### DISCUSSION THREAD

**Discussion by rohnnie** [community/community#32332] [Pull Requests]
I am trying to automate the process of raising a pr, waiting for the checks to be passed, and then merging via python code. I have used GitHub HTTPS Merge API to merge once the checks have been passed but im getting the following response. 

405 Method Not Allowed if merge cannot be performed 

I am able to successfully retrieve the pull number but not able to merge even if all the checks have been passed. 

Could anyone please suggest what could be the reason?

**Comment by airtower-luna** [comment]
My first guess would be a branch protection rule where some conditions aren't fulfilled, or maybe insufficient permissions on the token you use. But to be sure you'd have to look at the PR itself.

**Reply by rohnnie** [reply]
In Branch protection rule we have given the role to bypass branch protection rules and from token perspective wehave provided full permissions.
